src/Components/Components/src/Sections/SectionRegistry.cs

```diff
@@ -26,18 +26,18 @@ public void AddProvider(string name, ISectionContentProvider provider, bool isDe
         }
     }
 
-    public void RemoveProvider(string name, ISectionContentProvider provider)
+    public void RemoveProvider(object sectionId, ISectionContentProvider provider)
     {
-        if (!_providersByName.TryGetValue(name, out var providers))
+        if (!_providersBySectionId.TryGetValue(sectionId, out var providers))
         {
-            throw new InvalidOperationException($"There are no content providers with the name '{name}'.");
+            throw new InvalidOperationException($"There are no content providers with the given SectionId.");
```

Comment:
For debugging purposes, it might be nice to state what the value of `sectionId` was when this failed.
```suggestion
            throw new InvalidOperationException($"There are no content providers with the given section ID '{sectionId}'.");
```
But tbh this is a minor detail, so I'll leave it up to you in case you disagree. If you don't take that change, could we at least update it to:
```suggestion
            throw new InvalidOperationException($"There are no content providers with the given {nameof(SectionId)}.");
```
That way, we don't risk forgetting to update the exception message if we rename `SectionId` in the future 🙂

Reply:
Oh, and please apply changes to similar exception messages in this file to match whatever you decide to do here. Thanks!

Reply:
I have thought about it. 

The first one probably won't be very helpful since `SectionId` is object. The exception would be "There are no content providers with the given section ID 'System.Object'."

With the second one the problem is that `SectionRegistry` doesn't know about `SectionOutlet` or `SectionContent`. It works with `ISectionContentSubscriber` and `ISectionContentProvider` which makes sense. There is no `SectionId` in those interfaces. 
We could use `{nameof(SectionOutlet.SectionId)}` (depending on it is subscriber or provider) which I am fine with since `SectionRegistry` is internal. 
Or we could use `{nameof(sectionId)}` the name of the variable. This would not be of much help in case we rename `SectionId` in `SectionOutlet` and `SectionContent` we still need to remember to change the variable's name in `SectionRegistry` too.

What do you think?

Reply:
> "There are no content providers with the given section ID 'System.Object'."

It should still look fine if the user provides a value with a proper `ToString()` implementation (like a `string`, etc.). But I agree it can be confusing in some cases. Totally fine if you prefer to not include the value in the exception message, especially considering this is an exception that most likely indicate something incorrect about the _internal_ implementation, not the way the feature is being used. So, it's unlikely a customer will ever see it anyway (our code is perfect 😇).

> We could use {nameof(SectionOutlet.SectionId)}

Yeah, that seems good. Maybe use `SectionContent`'s instead simply because the exception message refers to a "content provider" which implies `SectionContent`.

_Or_... we could just use the phrase "section ID". That way it refers to the _concept_ of a section ID and not a literal symbol in code (the string "SectionId" implies there is some sort of connection to a variable, which is why I initially suggested using `nameof`).

Reply:
>we could just use the phrase "section ID". That way it refers to the concept

I like this option